### DISCUSSION TITLE
Multiple dims

### DISCUSSION THREAD

**Discussion by cvaldezerea** [JanMarvin/openxlsx2#546] [General]
Hi, another question, if I want to change format like in these cells A1:A10 and A11:A20 and A:30:A40, like jumping.  Another question in openxlsx, there was an option stack, how can I use it with add_numfmt or bold and color?

**Comment by JanMarvin** [comment]
Not sure if I understand you correctly, but maybe you are looking for something like this:

```R
wb <- wb_workbook()$add_worksheet()$
  add_data(x = rep(1, 30))$
  add_numfmt(dims = "A1:A10", numfmt = "0.0%")$
  add_font(dims = "A1:A10", name = "Arial", color = wb_color("red"), bold = "1")$
  add_numfmt(dims = "A11:A20", numfmt = "0.0")$
  add_font(dims = "A11:A20", name = "Arial", color = wb_color("green"), italic = "1")$
  add_numfmt(dims = "A21:A30", numfmt = "0.0")$
  add_font(dims = "A21:A30", name = "Arial", color = wb_color("blue"), strike = "1")
```

**Reply by cvaldezerea** [reply]
Yes, but is it possible to something like this
````
wb <- wb_workbook()$add_worksheet()$
  add_data(x = rep(1, 30))$
  add_numfmt(dims = c("A1:A10","A11:A20", "A21:A30"), numfmt = "0.0%")$
  add_font(dims = c("A1:A10","A11:A20", "A21:A30") name = "Arial", color = wb_color("red"), bold = "1")
````

**Reply by JanMarvin** [reply]
In `openxlsx2` styles stack per default. But when writing data you can decide if you want the cell style of the existing cell to remain (see `removeCellStyle` or if a style should be applied at all `applyCellStyle`).

Anyhow, you are living on the other side of the globe, it is quarter past midnight where I live and I get some sleep now. Let me know if you have further questions 👍

**Reply by cvaldezerea** [reply]
Thank you for all! Sorry! Good night!

**Reply by JanMarvin** [reply]
You can use experiment with something like this `"A1:A10;C1:C10"`, but it only works if the left and the right begin and end on the same column. I would suggest to simply work with independent dimensions. Having multiple `dims` as in your pseudo code above makes things only a bit harder for us.

If you need it frequently, you can simply wrap it in your own function or simply loop over the various dimensions.

```R
wb <- wb_workbook()$add_worksheet()$
  add_data(x = matrix(1, 10, 3), colNames = FALSE)$
  add_numfmt(dims = "A1:A10;C1:C10", numfmt = "0.0%")$
  add_font(dims = "A1:A10;C1:C10", name = "Arial", color = wb_color("red"), bold = "1")
```
